linera-base/src/data_types.rs

```diff
@@ -720,6 +725,8 @@ pub enum OracleResponse {
     Service(Vec<u8>),
     /// The response from an HTTP POST request.
     Post(Vec<u8>),
+    /// An assertion oracle that passed.
+    Assert,
```

Comment:
Question: do we have to distinguish which assertion passed? Just to confirm, as long as execution is deterministic, we don't right? Or is there any way to use a different oracle query to change execution in a way that changes which assertions pass?

Reply:
Exactly, we don't, since it's deterministic. In theory we wouldn't even need an `OracleResponse` entry, but I think it's cleaner if every oracle use corresponds to one entry.

Reply:
Yes, exactly, that wouldn't work if we didn't make an entry:
https://github.com/linera-io/linera-protocol/blob/6bda408766aa3e57c9cdd63f3cabdc781ae66f12/linera-core/src/chain_worker/state.rs#L326